core/state_processor.go

```diff
@@ -80,6 +80,10 @@ func (p *StateProcessor) Process(block *types.Block, statedb *state.StateDB, cfg
 	if beaconRoot := block.BeaconRoot(); beaconRoot != nil {
 		ProcessBeaconBlockRoot(*beaconRoot, vmenv, statedb)
 	}
+	if p.config.IsPrague(block.Number(), block.Time()) {
+		// This should not underflow as genesis block is not processed.
```

Comment:
This sentence doesn't make sense (perhaps it did in an earlier version of the code?). There's nothing to underflow. 
A `block.ParentHash()` returns a `hash`. If genesis were a block, it'd still be bound to return a `hash`
```suggestion
```